pyuvsim/tests/test_analyticbeam.py

```diff
@@ -118,6 +119,44 @@ def test_airy_beam():
     nt.assert_true(np.allclose(interpolated_beam, expected_data))
 
 
+def test_uv_beam_widths():
+    diameter_m = 200.
+    beam = pyuvsim.AnalyticBeam('airy', diameter=diameter_m)
+    beam.peak_normalize()
+    beam.interpolation_function = 'az_za_simple'
+
+    Nfreqs = 20
+    freq_vals = np.linspace(100e6, 130e6, Nfreqs)
+    lams = 3e8 / freq_vals
+
+    Npix = 700
+    zmax = np.radians(40)  # Degrees
+    x, y = np.meshgrid(range(-Npix // 2, Npix // 2), range(-Npix // 2, Npix // 2))
+    r = np.sqrt(x**2 + y**2) / float(Npix)
+    zas = 2 * r * zmax
+    azs = np.arctan2(y, x)
+    interpolated_beam, interp_basis_vector = beam.interp(az_array=np.array(azs),
+                                                         za_array=np.array(zas),
+                                                         freq_array=np.array(freq_vals))
+
+    ebeam = (interpolated_beam[0, 0, 0, :, :])
+    ebeam = ebeam.reshape(Nfreqs, Npix, Npix)
+    beam_kern = np.fft.fft2(ebeam, axes=(1, 2))
+    beam_kern = np.fft.fftshift(beam_kern, axes=(1, 2))
+
+    upix = np.pi / (2 * zmax)   # 2*zmax = FoV
+    uvals = np.arange(-Npix // 2, Npix // 2) * upix
+    for i, bk in enumerate(beam_kern):
+        thresh = np.abs(bk[Npix // 2, Npix // 2]) * 0.01    # Cutoff at 1% of the center value in Fourier space.
+        points = np.sum(np.abs(bk) >= thresh)
+        area = np.sum(points) * upix**2
+        kern_radius = np.sqrt(area / np.pi)
+
+        # For each frequency, check kernel width is less than dish diameter in wavelengths
+        nt.assert_true(kern_radius < 2 * diameter_m / lams[i])
+        print(kern_radius, 2 * diameter_m / lams[i])
```

Comment:
not sure we want to keep this print statement. Also, I think we want a test that's a little stronger. I don't just want it to be less, I want it to be close (for some definition of close).

Reply:
I'm not sure how best to address this... the widths are close, and seem to scale properly, but are off by a non-negligible amount. For this test, 2*D is about a factor of 1.2 times larger than the beam kernel radius. It's pretty consistently larger by that factor, but I'm not sure where that's coming from.

Reply:
Hmm, that is not as small factor as I would expect.

Reply:
I'm really worried about this, can we talk about it on tomorrow's telecon?

Reply:
Agreed. I've been fiddling with the test a little, and it looks like the error depends on the dish diameter, but not on the field of view or resolution.